Installing-and-Configuring/configuring-keys.md

```diff
@@ -0,0 +1,165 @@
+ you're using a new security key, you'll need to generate your key files. If you
+ already have the public key stubs for your security key, please proceed to
+ [Adding your PGP key](#adding-your-pgp-key).
+
+![GPG Management menu]({{ site.baseurl }}/images/gpg_management.jpg)
+
+Insert your security key and USB drive into the device, then chose "Generate GPG
+ keys manually on a USB security token".
+
+When at the GPG prompt, go into "Admin" mode and generate a new key inside the
+ security key:
+
+```shell
+admin
+generate
+```
+
+This will prompt you for the admin pin (`12345678` by default for Yubikey) and
+ then the existing pin (`123456` default for Yubikey).  Follow the other prompts
+ and eventually you should have a key saved onto the USB drive.
+
+When complete, you will be returned to menu.
+
+There is some more info in the [GPG guide](/GPG).
+
+Adding your PGP key
+===
+
+Heads uses your own GPG key to sign updates and as a result it needs the
+key stored in the ROM image before flashing the full Heads ROM.
+
+Ensure your security key and the USB drive with your key are still inserted.
+ Select  "Add a GPG key to the running BIOS" to enter the GPG Management menu,
+ then "Add a GPG key to the running BIOS + reflash".  Follow the steps and your
+ GPG key will be added to the Heads rom.
+
+Once `flashrom` is complete, reboot and now you should now be back in the Heads
+ runtime. It should display a message that is is unable to unseal TOTP.
+
+Because the reproducible flash has an empty MRC cache, you need to
+reboot one more time so that the PCR values as they would be going
+forward.
+
+Configuring the TPM
+===
+
+![TOTP/HOTP missing]({{ site.baseurl }}/images/otp_error.jpg)
+
+There aren't very many good details on how to setup TPMs, so this section could
+ use some work.
+
+Taking ownership
+---
+
+If you've acquired the machine from elsewhere, you'll need to establish physical
+ presence, perform a force clear and take ownership with your own password.
+ Should the storage root key (SRK) be set to something other than the well-known
+ password?
+
+```shell
+tpm-reset
+```
+
+There is something weird with enabling, presence and disabling. Sometimes reboot
+ fixes the state.
+
+tpmtotp
+---
+
+![TPMTOTP QR code]({{ site.baseurl }}/images/TPMTOTP_QR_code.jpg)
+
+Once you own the TPM, run `seal-totp` to generate a random secret, seal it with
+ the current TPM PCR values and store the sealed value in the TPM's NVRAM. This
+ will generate a QR code that you can scan with your google authenticator or
+ [FreeOTP+](https://f-droid.org/en/packages/org.liberty.android.freeotpplus/)
+ application and use to validate that the boot block, rom stage and Linux
+ payload are un-altered.
+
+![TPMTOTP output]({{ site.baseurl }}/images/TPMTOTP_output.jpg)
+
+On the next boot, or if you run `unseal-totp`, the script will extract the
+ sealed blob from the NVRAM and the TPM will validate that the PCR values are as
```

Comment:
All of this is dealt in FbWhiptail and should be moved to a deprecated instructions or something, really. I do not know who supports generic-init anymore (vs gui-init codepath)